### PR TITLE
Add necessary website files

### PR DESCRIPTION
**Pull request by nayres**
This branch adds necessary website files to migrate content from terraform-testing-sdk to terraform-plugin-testing.

### CODE REVIEW

File: website/Makefile [hunk 1]
```diff
@@ -0,0 +1,60 @@
+######################################################
+# NOTE: This file is managed by the Digital Team's   #
+# Terraform configuration @ hashicorp/mktg-terraform #
+######################################################
+
+.DEFAULT_GOAL := website
+
+# Set the preview mode for the website shell to "developer" or "io"
+PREVIEW_MODE ?= developer
+REPO ?= terraform-plugin-sdk
```
Comment (BRKalow) [inline website/Makefile:10]:
```suggestion
REPO ?= terraform-plugin-testing
```

File: website/Makefile [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+PWD=$$(pwd)
+
+DOCKER_IMAGE="hashicorp/dev-portal"
+DOCKER_IMAGE_LOCAL="dev-portal-local"
+DOCKER_RUN_FLAGS=-it \
+		--publish "3000:3000" \
+		--rm \
+		--tty \
+		--volume "$(PWD)/docs:/app/docs" \
+		--volume "$(PWD)/img:/app/public" \
```
Comment (BRKalow) [inline website/Makefile:26]:
```suggestion
		--volume "$(PWD)/img:/app/public/img" \
```

File: website/data/plugin-testing-nav-data.json [hunk 3]
```diff
@@ -0,0 +1,150 @@
+[
```
Comment (BRKalow) [inline website/data/plugin-testing-nav-data.json:1]: I think we can either exclude this file or leave it empty for now 👍

File: website/scripts/website-build.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+######################################################
+
+# Repo which we are cloning and executing npm run build:deploy-preview within
+REPO_TO_CLONE=dev-portal
+# Set the subdirectory name for the base project
+PREVIEW_DIR=website-preview
+# The directory we want to clone the project into
+CLONE_DIR=website-preview
+# The product for which we are building the deploy preview
+PRODUCT=terraform-plugin-sdk
```
Comment (BRKalow) [inline website/scripts/website-build.sh:13]:
```suggestion
PRODUCT=terraform-plugin-testing
```

**Review by BRKalow — COMMENTED** [review]
@thiskevinwang Do we also need to adjust the root `Makefile`?

File: website/data/plugin-testing-nav-data.json (hunk 3 above)
Comment (bflad) [reply website/data/plugin-testing-nav-data.json:1]: Refer also to https://github.com/hashicorp/terraform-plugin-testing/pull/34

**Comment by thiskevinwang** [comment]
> @thiskevinwang Do we also need to adjust the root `Makefile`?

Yea, the root `Makefile` should have targets that call out to `website/Makefile`

Example: https://github.com/hashicorp/terraform-plugin-framework/blob/main/Makefile

**Comment by nayres** [comment]
> > @thiskevinwang Do we also need to adjust the root `Makefile`?
> 
> Yea, the root `Makefile` should have targets that call out to `website/Makefile`
> 
> Example: https://github.com/hashicorp/terraform-plugin-framework/blob/main/Makefile



> > @thiskevinwang Do we also need to adjust the root `Makefile`?
> 
> Yea, the root `Makefile` should have targets that call out to `website/Makefile`
> 
> Example: https://github.com/hashicorp/terraform-plugin-framework/blob/main/Makefile

I can make this update. Would I have to point to the website/Makefile in a specific way?

**Comment by thiskevinwang** [comment]
> I can make this update. Would I have to point to the website/Makefile in a specific way?

@nayres Checkout this Makefile which is already doing what we're after
- https://github.com/hashicorp/terraform-plugin-framework/blob/main/Makefile

This is an example of how to call Makefile targets nested under `./website/`

**Review by bflad — APPROVED** [review]
Website loads okay with `make website`, `make website/build-local`, and `make website/local` -- I think we need the terraform-website bits setup for wiring /terraform/plugin/testing URLs to verify this local content unless there's another way.

**Pull request merged**